### PR TITLE
Fix value error generated on is_scalar check

### PR DESCRIPTION
**Pull request by erilyth**
`is_scalar = shape is not None and not shape` raises a value error when shape is a scalar, "ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()"

### CODE REVIEW

File: tensorflow/python/ops/variable_scope.py [hunk 1]
```diff
@@ -300,7 +300,7 @@ def _true_getter(name, shape=None, dtype=dtypes.float32,  # pylint: disable=miss
                      initializer=None, regularizer=None, reuse=None,
                      trainable=True, collections=None, caching_device=None,
                      partitioner=None, validate_shape=True, use_resource=None):
-      is_scalar = shape is not None and not shape
+      is_scalar = shape is not None and not isinstance(shape, tuple)
```
Comment (jhseu) [inline tensorflow/python/ops/variable_scope.py]: `isinstance(shape, collections_lib.Sequence)`?

Comment (erilyth) [inline tensorflow/python/ops/variable_scope.py]: Yes, that would be better.

**Comment by martinwicke** [comment]
Jenkins, test this please.

Jonathan, I assume this is good to go pending tests?

**Review by jhseu — APPROVED** [review]
Yep.

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by martinwicke** [comment]
Rebuilding broken Mac test here: 
https://ci.tensorflow.org/job/tensorflow-pull-requests-mac/5317/console

**Comment by martinwicke** [comment]
These failures (in the Linux CPU Py3 build) seem related.

**Comment by drpngx** [comment]
@erilyth any luck with the test failures?

**Comment by drpngx** [comment]
@erilyth here's an example failure:

```
ERROR: testIrisMomentum (__main__.CustomOptimizer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/estimators_test.py", line 142, in testIrisMomentum
    classifier.fit(x_train, y_train, steps=400)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/python/util/deprecation.py", line 296, in new_func
    return func(*args, **kwargs)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 442, in fit
    SKCompat(self).fit(x, y, batch_size, steps, max_steps, monitors)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 1353, in fit
    monitors=all_monitors)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/python/util/deprecation.py", line 296, in new_func
    return func(*args, **kwargs)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 458, in fit
    loss = self._train_model(input_fn=input_fn, hooks=hooks)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 958, in _train_model
    model_fn_ops = self._get_train_ops(features, labels)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 1165, in _get_train_ops
    return self._call_model_fn(features, labels, model_fn_lib.ModeKeys.TRAIN)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/estimator.py", line 1136, in _call_model_fn
    model_fn_results = self._model_fn(features, labels, **kwargs)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/dnn.py", line 209, in _dnn_model_fn
    logits=logits)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head.py", line 1059, in create_model_fn_ops
    enable_centered_bias=self._enable_centered_bias)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head.py", line 649, in _create_model_fn_ops
    batch_size, loss_fn, weight_tensor)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/head.py", line 1911, in _train_op
    train_op = train_op_fn(loss)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/learn/python/learn/estimators/dnn.py", line 202, in _train_op_fn
    summaries=[])
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/contrib/layers/python/layers/optimizers.py", line 180, in optimize_loss
    initializer=init_ops.constant_initializer(learning_rate))
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/python/ops/variable_scope.py", line 1065, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/python/ops/variable_scope.py", line 962, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/python/ops/variable_scope.py", line 367, in get_variable
    validate_shape=validate_shape, use_resource=use_resource)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/python/ops/variable_scope.py", line 321, in _true_getter
    use_resource=use_resource)
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/tensorflow/contrib/learn/estimators_test.runfiles/org_tensorflow/tensorflow/python/ops/variable_scope.py", line 463, in _get_partitioned_variable
    "shape: %s" % shape)
ValueError: A partitioned Variable must have rank at least 1, shape: ()

----------------------------------------------------------------------
```

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by frankchn** [comment]
ping @erilyth on the test failures.

**Comment by jhseu** [comment]
Jenkins, test this please

**Pull request merged**